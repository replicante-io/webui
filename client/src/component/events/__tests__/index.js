'use strict';
//@flow

// NOTE: This is required for generators at runtime.
import 'regenerator-runtime/runtime.js';

import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import React from 'react';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Events from '../index';


const mockStore = configureStore([]);
const EVENTS = [{
  event: "AGENT_DOWN",
  payload: {"SOME": "DATA"},
  timestamp: "2018-04-28T17:57:24.540480643Z"
}, {
  event: "AGENT_RECOVER",
  payload: {"SOME": "DATA"},
  timestamp: "2018-04-28T17:50:15.170187929Z"
}];


describe('Events', () => {
  describe('index', () => {
    let store;

    beforeEach(() => {
      store = mockStore({
        events: {
          events: []
        }
      });
    });

    test('renders index', () => {
      const tree = renderer.create(
        <BrowserRouter>
          <Provider store={store}>
            <Events />
          </Provider>
        </BrowserRouter>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('renders with data', () => {
      let store = mockStore({
        events: {
          events: EVENTS
        }
      });
      const tree = renderer.create(
        <BrowserRouter>
          <Provider store={store}>
            <Events />
          </Provider>
        </BrowserRouter>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });
});
