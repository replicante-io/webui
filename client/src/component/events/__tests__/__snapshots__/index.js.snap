// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events index renders index 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        Events
      </div>
      <div
        className="card-body text-center"
      >
        <p
          className="card-text large-icon"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-chart-area fa-w-16 "
            data-icon="chart-area"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M500 384c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H12c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v308h436zM372.7 159.5L288 216l-85.3-113.7c-5.1-6.8-15.5-6.3-19.9 1L96 248v104h384l-89.9-187.8c-3.2-6.5-11.4-8.7-17.4-4.7z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </p>
        <p
          className="card-text"
        >
          No data available
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Events index renders with data 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        Events
      </div>
      <div
        className="card-body text-center"
      >
        <div
          className="row event-item"
        >
          <div
            className="col timestamp"
          >
            2018-04-28T17:57:24.540480643Z
          </div>
          <div
            className="col event"
          >
            AGENT_DOWN
          </div>
          <div
            className="col-6 payload codeblock"
          >
            <div
              className="react-json-view"
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0)",
                  "cursor": "default",
                  "fontFamily": "monospace",
                  "position": "relative",
                }
              }
            >
              <div
                className="pretty-json-container object-container"
              >
                <div
                  className="object-content"
                >
                  <div
                    className="object-key-val"
                  >
                    <span>
                      <span
                        onClick={[Function]}
                        style={
                          Object {
                            "cursor": "pointer",
                            "display": "inline-block",
                          }
                        }
                      >
                        <div
                          className="icon-container"
                          style={
                            Object {
                              "display": "inline-block",
                              "width": "17px",
                            }
                          }
                        >
                          <span
                            className="expanded-icon"
                          >
                            <svg
                              fill="currentColor"
                              style={
                                Object {
                                  "color": "#586e75",
                                  "height": "1em",
                                  "paddingLeft": "2px",
                                  "verticalAlign": "top",
                                  "width": "1em",
                                }
                              }
                              viewBox="0 0 15 15"
                            >
                              <path
                                d="M0 5l6 6 6-6z"
                              />
                            </svg>
                          </span>
                        </div>
                        <span />
                        <span
                          style={
                            Object {
                              "color": "#002b36",
                              "cursor": "pointer",
                              "display": "inline-block",
                              "fontWeight": "bold",
                            }
                          }
                        >
                          {
                        </span>
                      </span>
                      <div
                        className="object-meta-data"
                        onClick={[Function]}
                        style={
                          Object {
                            "display": "inline-block",
                            "padding": "0px 0px 0px 10px",
                          }
                        }
                      >
                        <span
                          className="object-size"
                          style={
                            Object {
                              "borderRadius": "3px",
                              "color": "rgba(0, 0, 0, 0.3)",
                              "cursor": "default",
                              "fontStyle": "italic",
                              "margin": "0px 6px 0px 0px",
                            }
                          }
                        >
                          1
                           item
                          
                        </span>
                        <span
                          className="copy-to-clipboard-container"
                        >
                          <span
                            onClick={[Function]}
                            style={
                              Object {
                                "cursor": "pointer",
                                "display": "inline",
                              }
                            }
                          >
                            <span
                              className="copy-icon"
                            >
                              <svg
                                fill="currentColor"
                                preserveAspectRatio="xMidYMid meet"
                                style={
                                  Object {
                                    "color": "#268bd2",
                                    "fontSize": "15px",
                                    "height": "1em",
                                    "marginRight": "3px",
                                    "verticalAlign": "top",
                                    "width": "1em",
                                  }
                                }
                                viewBox="0 0 40 40"
                              >
                                <g>
                                  <path
                                    d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                                  />
                                </g>
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </span>
                    <div
                      className="pushed-content object-container"
                    >
                      <div
                        className="object-content"
                        style={
                          Object {
                            "marginLeft": "6px",
                          }
                        }
                      >
                        <div
                          className="variable-row"
                          style={
                            Object {
                              ":hover": Object {
                                "borderLeft": "2px solid rgb(235, 235, 235)",
                                "paddingLeft": "19px",
                              },
                              "borderLeft": "1px solid rgb(235, 235, 235)",
                              "paddingBottom": "3px",
                              "paddingLeft": 20,
                              "paddingRight": "5px",
                              "paddingTop": "3px",
                            }
                          }
                        >
                          <span>
                            <span
                              className="object-key"
                              style={
                                Object {
                                  ":hover": Object {
                                    "opacity": "1",
                                  },
                                  "color": "#002b36",
                                  "display": "inline-block",
                                  "fontStyle": "none",
                                  "letterSpacing": "0.5px",
                                  "opacity": "0.85",
                                  "verticalAlign": "top",
                                }
                              }
                            >
                              <span
                                style={
                                  Object {
                                    "verticalAlign": "top",
                                  }
                                }
                              >
                                "
                              </span>
                              <span
                                style={
                                  Object {
                                    "display": "inline-block",
                                  }
                                }
                              >
                                SOME
                              </span>
                              <span
                                style={
                                  Object {
                                    "verticalAlign": "top",
                                  }
                                }
                              >
                                "
                              </span>
                            </span>
                            <span
                              style={
                                Object {
                                  "color": "#002b36",
                                  "display": "inline-block",
                                  "margin": "0px 5px",
                                  "verticalAlign": "top",
                                }
                              }
                            >
                              :
                            </span>
                          </span>
                          <div
                            className="variable-value"
                            onClick={null}
                            style={
                              Object {
                                "cursor": "default",
                                "display": "inline-block",
                                "paddingRight": "6px",
                                "position": "relative",
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "color": "#cb4b16",
                                  "display": "inline-block",
                                }
                              }
                            >
                              <span
                                className="data-type-label"
                                style={
                                  Object {
                                    "fontSize": "11px",
                                    "marginRight": "4px",
                                    "opacity": "0.8",
                                  }
                                }
                              >
                                string
                              </span>
                              <span
                                className="string-value"
                                onClick={[Function]}
                                style={
                                  Object {
                                    "cursor": "default",
                                  }
                                }
                              >
                                "
                                DATA
                                "
                              </span>
                            </div>
                          </div>
                          <span
                            className="copy-to-clipboard-container"
                          >
                            <span
                              onClick={[Function]}
                              style={
                                Object {
                                  "cursor": "pointer",
                                  "display": "inline",
                                }
                              }
                            >
                              <span
                                className="copy-icon"
                              >
                                <svg
                                  fill="currentColor"
                                  preserveAspectRatio="xMidYMid meet"
                                  style={
                                    Object {
                                      "color": "#268bd2",
                                      "fontSize": "15px",
                                      "height": "1em",
                                      "marginRight": "3px",
                                      "verticalAlign": "top",
                                      "width": "1em",
                                    }
                                  }
                                  viewBox="0 0 40 40"
                                >
                                  <g>
                                    <path
                                      d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                                    />
                                  </g>
                                </svg>
                              </span>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                    <span
                      className="brace-row"
                    >
                      <span
                        style={
                          Object {
                            "color": "#002b36",
                            "cursor": "pointer",
                            "display": "inline-block",
                            "fontWeight": "bold",
                            "paddingLeft": "3px",
                          }
                        }
                      >
                        }
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="row event-item"
        >
          <div
            className="col timestamp"
          >
            2018-04-28T17:50:15.170187929Z
          </div>
          <div
            className="col event"
          >
            AGENT_RECOVER
          </div>
          <div
            className="col-6 payload codeblock"
          >
            <div
              className="react-json-view"
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0)",
                  "cursor": "default",
                  "fontFamily": "monospace",
                  "position": "relative",
                }
              }
            >
              <div
                className="pretty-json-container object-container"
              >
                <div
                  className="object-content"
                >
                  <div
                    className="object-key-val"
                  >
                    <span>
                      <span
                        onClick={[Function]}
                        style={
                          Object {
                            "cursor": "pointer",
                            "display": "inline-block",
                          }
                        }
                      >
                        <div
                          className="icon-container"
                          style={
                            Object {
                              "display": "inline-block",
                              "width": "17px",
                            }
                          }
                        >
                          <span
                            className="expanded-icon"
                          >
                            <svg
                              fill="currentColor"
                              style={
                                Object {
                                  "color": "#586e75",
                                  "height": "1em",
                                  "paddingLeft": "2px",
                                  "verticalAlign": "top",
                                  "width": "1em",
                                }
                              }
                              viewBox="0 0 15 15"
                            >
                              <path
                                d="M0 5l6 6 6-6z"
                              />
                            </svg>
                          </span>
                        </div>
                        <span />
                        <span
                          style={
                            Object {
                              "color": "#002b36",
                              "cursor": "pointer",
                              "display": "inline-block",
                              "fontWeight": "bold",
                            }
                          }
                        >
                          {
                        </span>
                      </span>
                      <div
                        className="object-meta-data"
                        onClick={[Function]}
                        style={
                          Object {
                            "display": "inline-block",
                            "padding": "0px 0px 0px 10px",
                          }
                        }
                      >
                        <span
                          className="object-size"
                          style={
                            Object {
                              "borderRadius": "3px",
                              "color": "rgba(0, 0, 0, 0.3)",
                              "cursor": "default",
                              "fontStyle": "italic",
                              "margin": "0px 6px 0px 0px",
                            }
                          }
                        >
                          1
                           item
                          
                        </span>
                        <span
                          className="copy-to-clipboard-container"
                        >
                          <span
                            onClick={[Function]}
                            style={
                              Object {
                                "cursor": "pointer",
                                "display": "inline",
                              }
                            }
                          >
                            <span
                              className="copy-icon"
                            >
                              <svg
                                fill="currentColor"
                                preserveAspectRatio="xMidYMid meet"
                                style={
                                  Object {
                                    "color": "#268bd2",
                                    "fontSize": "15px",
                                    "height": "1em",
                                    "marginRight": "3px",
                                    "verticalAlign": "top",
                                    "width": "1em",
                                  }
                                }
                                viewBox="0 0 40 40"
                              >
                                <g>
                                  <path
                                    d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                                  />
                                </g>
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </span>
                    <div
                      className="pushed-content object-container"
                    >
                      <div
                        className="object-content"
                        style={
                          Object {
                            "marginLeft": "6px",
                          }
                        }
                      >
                        <div
                          className="variable-row"
                          style={
                            Object {
                              ":hover": Object {
                                "borderLeft": "2px solid rgb(235, 235, 235)",
                                "paddingLeft": "19px",
                              },
                              "borderLeft": "1px solid rgb(235, 235, 235)",
                              "paddingBottom": "3px",
                              "paddingLeft": 20,
                              "paddingRight": "5px",
                              "paddingTop": "3px",
                            }
                          }
                        >
                          <span>
                            <span
                              className="object-key"
                              style={
                                Object {
                                  ":hover": Object {
                                    "opacity": "1",
                                  },
                                  "color": "#002b36",
                                  "display": "inline-block",
                                  "fontStyle": "none",
                                  "letterSpacing": "0.5px",
                                  "opacity": "0.85",
                                  "verticalAlign": "top",
                                }
                              }
                            >
                              <span
                                style={
                                  Object {
                                    "verticalAlign": "top",
                                  }
                                }
                              >
                                "
                              </span>
                              <span
                                style={
                                  Object {
                                    "display": "inline-block",
                                  }
                                }
                              >
                                SOME
                              </span>
                              <span
                                style={
                                  Object {
                                    "verticalAlign": "top",
                                  }
                                }
                              >
                                "
                              </span>
                            </span>
                            <span
                              style={
                                Object {
                                  "color": "#002b36",
                                  "display": "inline-block",
                                  "margin": "0px 5px",
                                  "verticalAlign": "top",
                                }
                              }
                            >
                              :
                            </span>
                          </span>
                          <div
                            className="variable-value"
                            onClick={null}
                            style={
                              Object {
                                "cursor": "default",
                                "display": "inline-block",
                                "paddingRight": "6px",
                                "position": "relative",
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "color": "#cb4b16",
                                  "display": "inline-block",
                                }
                              }
                            >
                              <span
                                className="data-type-label"
                                style={
                                  Object {
                                    "fontSize": "11px",
                                    "marginRight": "4px",
                                    "opacity": "0.8",
                                  }
                                }
                              >
                                string
                              </span>
                              <span
                                className="string-value"
                                onClick={[Function]}
                                style={
                                  Object {
                                    "cursor": "default",
                                  }
                                }
                              >
                                "
                                DATA
                                "
                              </span>
                            </div>
                          </div>
                          <span
                            className="copy-to-clipboard-container"
                          >
                            <span
                              onClick={[Function]}
                              style={
                                Object {
                                  "cursor": "pointer",
                                  "display": "inline",
                                }
                              }
                            >
                              <span
                                className="copy-icon"
                              >
                                <svg
                                  fill="currentColor"
                                  preserveAspectRatio="xMidYMid meet"
                                  style={
                                    Object {
                                      "color": "#268bd2",
                                      "fontSize": "15px",
                                      "height": "1em",
                                      "marginRight": "3px",
                                      "verticalAlign": "top",
                                      "width": "1em",
                                    }
                                  }
                                  viewBox="0 0 40 40"
                                >
                                  <g>
                                    <path
                                      d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                                    />
                                  </g>
                                </svg>
                              </span>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                    <span
                      className="brace-row"
                    >
                      <span
                        style={
                          Object {
                            "color": "#002b36",
                            "cursor": "pointer",
                            "display": "inline-block",
                            "fontWeight": "bold",
                            "paddingLeft": "3px",
                          }
                        }
                      >
                        }
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
